pkg/controller/priorityqueue/priorityqueue_test.go

```diff
@@ -197,6 +197,66 @@ var _ = Describe("Controllerworkqueue", func() {
 		Expect(metrics.retries["test"]).To(Equal(1))
 	})
 
+	It("returns high priority item that became ready before low priority items", func() {
+		q, metrics := newQueue()
+		defer q.ShutDown()
+
+		now := time.Now().Round(time.Second)
+		nowLock := sync.Mutex{}
+		tick := make(chan time.Time)
+
+		cwq := q.(*priorityqueue[string])
+		cwq.now = func() time.Time {
+			nowLock.Lock()
+			defer nowLock.Unlock()
+			return now
+		}
+		cwq.tick = func(d time.Duration) <-chan time.Time {
+			Expect(d).To(Equal(time.Second))
+			return tick
+		}
+
+		retrievedItem := make(chan any)
+		getNext := make(chan any)
+
+		go func() {
+			defer GinkgoRecover()
+			defer close(retrievedItem)
+
+			key, prio, _ := q.GetWithPriority()
+			Expect(key).To(Equal("foo"))
+			Expect(prio).To(Equal(-100))
+
+			retrievedItem <- nil
+			<-getNext
+
+			key, prio, _ = q.GetWithPriority()
+			Expect(key).To(Equal("prio"))
+			Expect(prio).To(Equal(0))
+		}()
+
+		lowPriority := -100
+		highPriority := 0
+		q.AddWithOpts(AddOpts{After: 0, Priority: &lowPriority}, "foo")
+		q.AddWithOpts(AddOpts{After: 0, Priority: &lowPriority}, "bar")
+		q.AddWithOpts(AddOpts{After: time.Second, Priority: &highPriority}, "prio")
```

Comment:
I think youn can simplify this test by removing the asynchronous part from the test and just do the `now.Add(time.Second)` right after this, then do a `Get` and validate that that returns the high prio item